src/Helpers.php

```diff
@@ -275,6 +275,16 @@ public static function collectUsedRefs(array $data): array {
 		return $refs;
 	}
 
+	public static function mergeUsedRefs(array $usedSchemas, array $target): array {
```

Comment:
Instead of passing the usedSchemas to the method just do the merging outside of the method

Reply:
This will will duplicate the merge because we need to have the merge inside method.

Reply:
No, the method can return an array itself and then you do the merging using `array_merge($usedSchemas, ...self::mergeUsedRefs())`

Reply:
Could you check again?